hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/test/meta/SigTransactionMetadataBuilderTest.java

```diff
@@ -0,0 +1,325 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.test.meta;
+
+import static com.hedera.node.app.spi.KeyOrLookupFailureReason.withFailureReason;
+import static com.hedera.node.app.spi.KeyOrLookupFailureReason.withKey;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+
+import com.google.protobuf.ByteString;
+import com.hedera.node.app.spi.AccountKeyLookup;
+import com.hedera.node.app.spi.KeyOrLookupFailureReason;
+import com.hedera.node.app.spi.key.HederaKey;
+import com.hedera.node.app.spi.meta.SigTransactionMetadataBuilder;
+import com.hedera.node.app.spi.meta.TransactionMetadata;
+import com.hederahashgraph.api.proto.java.*;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class SigTransactionMetadataBuilderTest {
+    public static final com.hederahashgraph.api.proto.java.Key A_COMPLEX_KEY =
+            com.hederahashgraph.api.proto.java.Key.newBuilder()
+                    .setThresholdKey(
+                            ThresholdKey.newBuilder()
+                                    .setThreshold(2)
+                                    .setKeys(
+                                            KeyList.newBuilder()
+                                                    .addKeys(
+                                                            com.hederahashgraph.api.proto.java.Key
+                                                                    .newBuilder()
+                                                                    .setEd25519(
+                                                                            ByteString.copyFrom(
+                                                                                    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+                                                                                            .getBytes())))
+                                                    .addKeys(
+                                                            com.hederahashgraph.api.proto.java.Key
+                                                                    .newBuilder()
+                                                                    .setEd25519(
+                                                                            ByteString.copyFrom(
+                                                                                    "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+                                                                                            .getBytes())))))
```

Comment:
Ugh. I hate this spotless formatting. One of these days I'm going to lose my patience and write a better spotless plugin that formats things nicely :)

Reply:
💯